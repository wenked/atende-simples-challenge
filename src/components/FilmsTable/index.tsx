import React from 'react';
import { filmInterface } from 'src/types';
import { Cell, Container } from './styles';

interface FilmsTableProps {
	filmsData: filmInterface[];
}

const FilmsTable: React.FC<FilmsTableProps> = ({ filmsData }) => {
	return (
		<Container>
			<Cell id='row-header'>Episode Id</Cell>
			<Cell id='row-header'>ID</Cell>
			<Cell id='row-header'>Title</Cell>
			<Cell id='row-header'>Director</Cell>
			<Cell id='row-header'>Opening Crawl</Cell>
			<Cell id='row-header'>Producers</Cell>
			<Cell id='row-header'>Release Date</Cell>
			{filmsData.map((film, i) => {
				return (
					<React.Fragment key={i}>
						<Cell data-title='Episode ID'>{film.episodeID}</Cell>
						<Cell data-title='ID'>{film.id}</Cell>
						<Cell data-title='Title'>{film.title}</Cell>
						<Cell data-title='Director'>{film.director}</Cell>
						<Cell data-title='Opening Crawl'>{film.openingCrawl}</Cell>
						<Cell data-title='Producers'>
							{film.producers.map((producer, i) => {
								return (
									<React.Fragment key={i}>
										{producer}
										<br />
									</React.Fragment>
								);
							})}
						</Cell>
						<Cell data-title='Release Date'>{film.releaseDate}</Cell>
					</React.Fragment>
				);
			})}
		</Container>
	);
};

export default FilmsTable;
