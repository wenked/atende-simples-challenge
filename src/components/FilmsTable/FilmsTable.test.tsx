import React from 'react';
import { mount } from 'enzyme';
import { MockedProvider } from '@apollo/client/testing';
import FilmsTable from './index';
import { render } from '@testing-library/react';

const films = [
	{
		id: 'ZmlsbXM6MQ==',
		episodeID: 4,
		title: 'A New Hope',
		openingCrawl:
			"It is a period of civil war.\r\nRebel spaceships, striking\r\nfrom a hidden base, have won\r\ntheir first victory against\r\nthe evil Galactic Empire.\r\n\r\nDuring the battle, Rebel\r\nspies managed to steal secret\r\nplans to the Empire's\r\nultimate weapon, the DEATH\r\nSTAR, an armored space\r\nstation with enough power\r\nto destroy an entire planet.\r\n\r\nPursued by the Empire's\r\nsinister agents, Princess\r\nLeia races home aboard her\r\nstarship, custodian of the\r\nstolen plans that can save her\r\npeople and restore\r\nfreedom to the galaxy....",
		director: 'George Lucas',
		producers: ['Gary Kurtz', 'Rick McCallum'],
		releaseDate: '1977-05-25',
		created: '2014-12-10T14:23:31.880000Z',
		edited: '2014-12-20T19:49:45.256000Z',
	},
	{
		id: 'ZmlsbXM6Mg==',
		episodeID: 5,
		title: 'The Empire Strikes Back',
		openingCrawl:
			'It is a dark time for the\r\nRebellion. Although the Death\r\nStar has been destroyed,\r\nImperial troops have driven the\r\nRebel forces from their hidden\r\nbase and pursued them across\r\nthe galaxy.\r\n\r\nEvading the dreaded Imperial\r\nStarfleet, a group of freedom\r\nfighters led by Luke Skywalker\r\nhas established a new secret\r\nbase on the remote ice world\r\nof Hoth.\r\n\r\nThe evil lord Darth Vader,\r\nobsessed with finding young\r\nSkywalker, has dispatched\r\nthousands of remote probes into\r\nthe far reaches of space....',
		director: 'Irvin Kershner',
		producers: ['Gary Kurtz', 'Rick McCallum'],
		releaseDate: '1980-05-17',
		created: '2014-12-12T11:26:24.656000Z',
		edited: '2014-12-15T13:07:53.386000Z',
	},
	{
		id: 'ZmlsbXM6Mw==',
		episodeID: 6,
		title: 'Return of the Jedi',
		openingCrawl:
			'Luke Skywalker has returned to\r\nhis home planet of Tatooine in\r\nan attempt to rescue his\r\nfriend Han Solo from the\r\nclutches of the vile gangster\r\nJabba the Hutt.\r\n\r\nLittle does Luke know that the\r\nGALACTIC EMPIRE has secretly\r\nbegun construction on a new\r\narmored space station even\r\nmore powerful than the first\r\ndreaded Death Star.\r\n\r\nWhen completed, this ultimate\r\nweapon will spell certain doom\r\nfor the small band of rebels\r\nstruggling to restore freedom\r\nto the galaxy...',
		director: 'Richard Marquand',
		producers: ['Howard G. Kazanjian', 'George Lucas', 'Rick McCallum'],
		releaseDate: '1983-05-25',
		created: '2014-12-18T10:39:33.255000Z',
		edited: '2014-12-20T09:48:37.462000Z',
	},
	{
		id: 'ZmlsbXM6NA==',
		episodeID: 1,
		title: 'The Phantom Menace',
		openingCrawl:
			'Turmoil has engulfed the\r\nGalactic Republic. The taxation\r\nof trade routes to outlying star\r\nsystems is in dispute.\r\n\r\nHoping to resolve the matter\r\nwith a blockade of deadly\r\nbattleships, the greedy Trade\r\nFederation has stopped all\r\nshipping to the small planet\r\nof Naboo.\r\n\r\nWhile the Congress of the\r\nRepublic endlessly debates\r\nthis alarming chain of events,\r\nthe Supreme Chancellor has\r\nsecretly dispatched two Jedi\r\nKnights, the guardians of\r\npeace and justice in the\r\ngalaxy, to settle the conflict....',
		director: 'George Lucas',
		producers: ['Rick McCallum'],
		releaseDate: '1999-05-19',
		created: '2014-12-19T16:52:55.740000Z',
		edited: '2014-12-20T10:54:07.216000Z',
	},
	{
		id: 'ZmlsbXM6NQ==',
		episodeID: 2,
		title: 'Attack of the Clones',
		openingCrawl:
			'There is unrest in the Galactic\r\nSenate. Several thousand solar\r\nsystems have declared their\r\nintentions to leave the Republic.\r\n\r\nThis separatist movement,\r\nunder the leadership of the\r\nmysterious Count Dooku, has\r\nmade it difficult for the limited\r\nnumber of Jedi Knights to maintain \r\npeace and order in the galaxy.\r\n\r\nSenator Amidala, the former\r\nQueen of Naboo, is returning\r\nto the Galactic Senate to vote\r\non the critical issue of creating\r\nan ARMY OF THE REPUBLIC\r\nto assist the overwhelmed\r\nJedi....',
		director: 'George Lucas',
		producers: ['Rick McCallum'],
		releaseDate: '2002-05-16',
		created: '2014-12-20T10:57:57.886000Z',
		edited: '2014-12-20T20:18:48.516000Z',
	},
	{
		id: 'ZmlsbXM6Ng==',
		episodeID: 3,
		title: 'Revenge of the Sith',
		openingCrawl:
			'War! The Republic is crumbling\r\nunder attacks by the ruthless\r\nSith Lord, Count Dooku.\r\nThere are heroes on both sides.\r\nEvil is everywhere.\r\n\r\nIn a stunning move, the\r\nfiendish droid leader, General\r\nGrievous, has swept into the\r\nRepublic capital and kidnapped\r\nChancellor Palpatine, leader of\r\nthe Galactic Senate.\r\n\r\nAs the Separatist Droid Army\r\nattempts to flee the besieged\r\ncapital with their valuable\r\nhostage, two Jedi Knights lead a\r\ndesperate mission to rescue the\r\ncaptive Chancellor....',
		director: 'George Lucas',
		producers: ['Rick McCallum'],
		releaseDate: '2005-05-19',
		created: '2014-12-20T18:49:38.403000Z',
		edited: '2014-12-20T20:47:52.073000Z',
	},
];

describe('rendering component', () => {
	it('rendering table without crashing, and checking data', () => {
		const wrapper = mount(
			<MockedProvider>
				<FilmsTable filmsData={films} />
			</MockedProvider>
		);
		console.log(wrapper.debug());
	});
});

test('snapshot', () => {
	const wrapper = render(
		<MockedProvider>
			<FilmsTable filmsData={films} />
		</MockedProvider>
	);
	expect(wrapper.container.firstChild).toMatchSnapshot();
});
