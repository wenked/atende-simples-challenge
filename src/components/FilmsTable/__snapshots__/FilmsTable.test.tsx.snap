// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<div
  class="sc-bdfBwQ gzySuu"
>
  <div
    class="sc-gsTCUz jnIFRw"
    id="row-header"
  >
    Episode Id
  </div>
  <div
    class="sc-gsTCUz jnIFRw"
    id="row-header"
  >
    ID
  </div>
  <div
    class="sc-gsTCUz jnIFRw"
    id="row-header"
  >
    Title
  </div>
  <div
    class="sc-gsTCUz jnIFRw"
    id="row-header"
  >
    Director
  </div>
  <div
    class="sc-gsTCUz jnIFRw"
    id="row-header"
  >
    Opening Crawl
  </div>
  <div
    class="sc-gsTCUz jnIFRw"
    id="row-header"
  >
    Producers
  </div>
  <div
    class="sc-gsTCUz jnIFRw"
    id="row-header"
  >
    Release Date
  </div>
  <div
    class="sc-gsTCUz jnIFRw"
    data-title="Episode ID"
  >
    4
  </div>
  <div
    class="sc-gsTCUz jnIFRw"
    data-title="ID"
  >
    ZmlsbXM6MQ==
  </div>
  <div
    class="sc-gsTCUz jnIFRw"
    data-title="Title"
  >
    A New Hope
  </div>
  <div
    class="sc-gsTCUz jnIFRw"
    data-title="Director"
  >
    George Lucas
  </div>
  <div
    class="sc-gsTCUz jnIFRw"
    data-title="Opening Crawl"
  >
    It is a period of civil war.
Rebel spaceships, striking
from a hidden base, have won
their first victory against
the evil Galactic Empire.

During the battle, Rebel
spies managed to steal secret
plans to the Empire's
ultimate weapon, the DEATH
STAR, an armored space
station with enough power
to destroy an entire planet.

Pursued by the Empire's
sinister agents, Princess
Leia races home aboard her
starship, custodian of the
stolen plans that can save her
people and restore
freedom to the galaxy....
  </div>
  <div
    class="sc-gsTCUz jnIFRw"
    data-title="Producers"
  >
    Gary Kurtz
    <br />
    Rick McCallum
    <br />
  </div>
  <div
    class="sc-gsTCUz jnIFRw"
    data-title="Release Date"
  >
    1977-05-25
  </div>
  <div
    class="sc-gsTCUz jnIFRw"
    data-title="Episode ID"
  >
    5
  </div>
  <div
    class="sc-gsTCUz jnIFRw"
    data-title="ID"
  >
    ZmlsbXM6Mg==
  </div>
  <div
    class="sc-gsTCUz jnIFRw"
    data-title="Title"
  >
    The Empire Strikes Back
  </div>
  <div
    class="sc-gsTCUz jnIFRw"
    data-title="Director"
  >
    Irvin Kershner
  </div>
  <div
    class="sc-gsTCUz jnIFRw"
    data-title="Opening Crawl"
  >
    It is a dark time for the
Rebellion. Although the Death
Star has been destroyed,
Imperial troops have driven the
Rebel forces from their hidden
base and pursued them across
the galaxy.

Evading the dreaded Imperial
Starfleet, a group of freedom
fighters led by Luke Skywalker
has established a new secret
base on the remote ice world
of Hoth.

The evil lord Darth Vader,
obsessed with finding young
Skywalker, has dispatched
thousands of remote probes into
the far reaches of space....
  </div>
  <div
    class="sc-gsTCUz jnIFRw"
    data-title="Producers"
  >
    Gary Kurtz
    <br />
    Rick McCallum
    <br />
  </div>
  <div
    class="sc-gsTCUz jnIFRw"
    data-title="Release Date"
  >
    1980-05-17
  </div>
  <div
    class="sc-gsTCUz jnIFRw"
    data-title="Episode ID"
  >
    6
  </div>
  <div
    class="sc-gsTCUz jnIFRw"
    data-title="ID"
  >
    ZmlsbXM6Mw==
  </div>
  <div
    class="sc-gsTCUz jnIFRw"
    data-title="Title"
  >
    Return of the Jedi
  </div>
  <div
    class="sc-gsTCUz jnIFRw"
    data-title="Director"
  >
    Richard Marquand
  </div>
  <div
    class="sc-gsTCUz jnIFRw"
    data-title="Opening Crawl"
  >
    Luke Skywalker has returned to
his home planet of Tatooine in
an attempt to rescue his
friend Han Solo from the
clutches of the vile gangster
Jabba the Hutt.

Little does Luke know that the
GALACTIC EMPIRE has secretly
begun construction on a new
armored space station even
more powerful than the first
dreaded Death Star.

When completed, this ultimate
weapon will spell certain doom
for the small band of rebels
struggling to restore freedom
to the galaxy...
  </div>
  <div
    class="sc-gsTCUz jnIFRw"
    data-title="Producers"
  >
    Howard G. Kazanjian
    <br />
    George Lucas
    <br />
    Rick McCallum
    <br />
  </div>
  <div
    class="sc-gsTCUz jnIFRw"
    data-title="Release Date"
  >
    1983-05-25
  </div>
  <div
    class="sc-gsTCUz jnIFRw"
    data-title="Episode ID"
  >
    1
  </div>
  <div
    class="sc-gsTCUz jnIFRw"
    data-title="ID"
  >
    ZmlsbXM6NA==
  </div>
  <div
    class="sc-gsTCUz jnIFRw"
    data-title="Title"
  >
    The Phantom Menace
  </div>
  <div
    class="sc-gsTCUz jnIFRw"
    data-title="Director"
  >
    George Lucas
  </div>
  <div
    class="sc-gsTCUz jnIFRw"
    data-title="Opening Crawl"
  >
    Turmoil has engulfed the
Galactic Republic. The taxation
of trade routes to outlying star
systems is in dispute.

Hoping to resolve the matter
with a blockade of deadly
battleships, the greedy Trade
Federation has stopped all
shipping to the small planet
of Naboo.

While the Congress of the
Republic endlessly debates
this alarming chain of events,
the Supreme Chancellor has
secretly dispatched two Jedi
Knights, the guardians of
peace and justice in the
galaxy, to settle the conflict....
  </div>
  <div
    class="sc-gsTCUz jnIFRw"
    data-title="Producers"
  >
    Rick McCallum
    <br />
  </div>
  <div
    class="sc-gsTCUz jnIFRw"
    data-title="Release Date"
  >
    1999-05-19
  </div>
  <div
    class="sc-gsTCUz jnIFRw"
    data-title="Episode ID"
  >
    2
  </div>
  <div
    class="sc-gsTCUz jnIFRw"
    data-title="ID"
  >
    ZmlsbXM6NQ==
  </div>
  <div
    class="sc-gsTCUz jnIFRw"
    data-title="Title"
  >
    Attack of the Clones
  </div>
  <div
    class="sc-gsTCUz jnIFRw"
    data-title="Director"
  >
    George Lucas
  </div>
  <div
    class="sc-gsTCUz jnIFRw"
    data-title="Opening Crawl"
  >
    There is unrest in the Galactic
Senate. Several thousand solar
systems have declared their
intentions to leave the Republic.

This separatist movement,
under the leadership of the
mysterious Count Dooku, has
made it difficult for the limited
number of Jedi Knights to maintain 
peace and order in the galaxy.

Senator Amidala, the former
Queen of Naboo, is returning
to the Galactic Senate to vote
on the critical issue of creating
an ARMY OF THE REPUBLIC
to assist the overwhelmed
Jedi....
  </div>
  <div
    class="sc-gsTCUz jnIFRw"
    data-title="Producers"
  >
    Rick McCallum
    <br />
  </div>
  <div
    class="sc-gsTCUz jnIFRw"
    data-title="Release Date"
  >
    2002-05-16
  </div>
  <div
    class="sc-gsTCUz jnIFRw"
    data-title="Episode ID"
  >
    3
  </div>
  <div
    class="sc-gsTCUz jnIFRw"
    data-title="ID"
  >
    ZmlsbXM6Ng==
  </div>
  <div
    class="sc-gsTCUz jnIFRw"
    data-title="Title"
  >
    Revenge of the Sith
  </div>
  <div
    class="sc-gsTCUz jnIFRw"
    data-title="Director"
  >
    George Lucas
  </div>
  <div
    class="sc-gsTCUz jnIFRw"
    data-title="Opening Crawl"
  >
    War! The Republic is crumbling
under attacks by the ruthless
Sith Lord, Count Dooku.
There are heroes on both sides.
Evil is everywhere.

In a stunning move, the
fiendish droid leader, General
Grievous, has swept into the
Republic capital and kidnapped
Chancellor Palpatine, leader of
the Galactic Senate.

As the Separatist Droid Army
attempts to flee the besieged
capital with their valuable
hostage, two Jedi Knights lead a
desperate mission to rescue the
captive Chancellor....
  </div>
  <div
    class="sc-gsTCUz jnIFRw"
    data-title="Producers"
  >
    Rick McCallum
    <br />
  </div>
  <div
    class="sc-gsTCUz jnIFRw"
    data-title="Release Date"
  >
    2005-05-19
  </div>
</div>
`;
